internal/pkg/composable/providers/host/host.go

```diff
@@ -114,6 +112,16 @@ func ContextProviderBuilder(log *logger.Logger, c *config.Config, _ bool) (corec
 	if p.CheckInterval <= 0 {
 		p.CheckInterval = DefaultCheckInterval
 	}
+
+	p.fqdnFFChangeCh = make(chan struct{})
```

Comment:
I would also turn this into a buffered channel so its not blocking when the callback hook is called but the provider is not running.

Now with it separate it is possible that no one is reading from the channel when it is pushed onto the channel.

Reply:
Thanks.  Addressed in b0b55c3b33cc4be77acd50a234c5340851f15593.